WordPress/Classes/ViewRelated/Reader/DetailWebView/OfflineReaderWebView.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+/// A WKWebView used to save post content to be available when offline
+/// The mechanism is quite simple: open the post in a hidden webview so the images are cached
+///
+class OfflineReaderWebView: ReaderWebView {
+    func saveForLater(_ post: ReaderPost, viewController: UIViewController) {
+        guard let contentForDisplay = post.contentForDisplay() else {
+            return
+        }
+
+        navigationDelegate = self
+
+        frame = CGRect(x: 0, y: 0, width: viewController.view.frame.width, height: viewController.view.frame.height)
+
+        isHidden = true
+
+        viewController.view.addSubview(self)
+
+        load(contentForDisplay)
+    }
+
+    private func load(_ string: String) {
+        // Remove all srcset from the images, only the URL in the src tag will be cached
+        let content = super.formattedContent(string, additionalJavaScript: """
+            document.querySelectorAll('img').forEach((el) => {el.removeAttribute('srcset')})
+        """)
+
+        super.loadHTMLString(content, baseURL: Bundle.wordPressSharedBundle.bundleURL)
+    }
+}
+
+extension OfflineReaderWebView: WKNavigationDelegate {
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        // We wait 5 secs before removing the WebView to give it time to be rendered
+        // If its removed before it was rendered, the images won't be cached
+        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
```

Comment:
I know the technical challenges with properly detecting whether all the images have successfully loaded in the WebView, but 5 seconds is a pretty long time.  Have you tested using a smaller time interval?

Reply:
My idea is that the webview might have different times to render from device to device, so even if I'm able to find a certain amount of seconds for the simulator or the iPhone 11, will that work for an iPhone 7?

5s it's a lot, but it gives me enough confidence that this would work for most of the cases.

Reply:
That makes sense. Plus it the total time will vary depending on network speeds as well. I think in the future we should try to come up with a better solution.